### PR TITLE
Add layers support

### PR DESCRIPTION
**Pull request by darlev91**
See Issue #1001 

Description of changes: Added layer support to Config and Lambda Function objects.

### CODE REVIEW

File: chalice/deploy/deployer.py [hunk 1]
```diff
@@ -606,6 +606,7 @@ def _build_lambda_function(self,
             security_group_ids=security_group_ids,
             subnet_ids=subnet_ids,
             reserved_concurrency=config.reserved_concurrency,
+            layers=config.layers
```
Comment (jamesls) [inline chalice/deploy/deployer.py]: We should add the same logic we have for subnet/security groups where we default to an empty list if no configuration is provided in `.chalice/config.json`.  One of the things I noticed while trying out this PR was that if I had a layer in my chalice config, deploy my app, then remove the layer from my config, the layer will still be part of the function configuration.

Comment (darlev91) [reply chalice/deploy/deployer.py]: @jamesls Thanks for your feedback!
We are using AWS layers to give the possibility to use dependencies of operating system unavailable in lambda functions by default.
For instance, we need to use git and ssh clients which is not installed in AMI.
We install git and ssh from source files and build it statically, after that we compressed
installed data and store it as lambda layers.
Some python libraries require dependencies that need to be installed in OS.
For example, GitPython requires executable git binary.

**Comment by stealthycoin** [comment]
Looks good to me.

**Comment by jpeters71** [comment]
So...how exactly is this supposed to work?  If chalice doesn't remove dependencies that are duplicated between the layer and what it has in the vendor dir (or resolved by the requirements.txt file), it makes the layer pointless, right?

I think we're missing something...

Here's the scenario: we have an API that's fairly simple, but has a lot of code behind it. We're running into the AWS lambda size limit (50mb). The biggest package we use is numpy. We add a reference to the numpy layer from our chalice config.json file and try to deploy.  Same problem as before because chalice doesn't look at the layer during the packaging phase so it doesn't know that it doesn't need to package up numpy.

We've tried a few things; like removing numpy from our requirements.txt file.  But, then, chalice won't package it because it can't load our app.py since, deep down in the code that gets pulled in, is a reference to numpy which python can't resolve.

So, where we're at now is: we're using chalice to generate the package with all the dependencies. Then, we manually unzip, remove the packages we know are in the layer (i.e. numpy); zip it back up; use awscli to deploy it.  That seems really cumbersome.  Again, I have to think we're missing something.

Is this the recommended process for doing this?

**Comment by skwashd** [comment]
@jpeters71 this is the initial implementation of layers support. Chalice attaches the layer/s to the function. 

In our case we use gitpython, which requires the `git` binary. A layer was the best way for us to do this. 

Issue #1001 discusses the next steps for how chalice can take full advantage of layers. The features being discussed there should give you the functionality you're after.

**Comment by vrinda1410** [comment]
@jamesls @stealthycoin Is there a tested & clean approach to this question : https://github.com/aws/chalice/pull/1063#issuecomment-516619913

My scenario is similar wherein I have bundled few functions from my code & published them as a lambda layer. e.g. DB Connection code
But I can't seem to call that as layer in my code as neither "chalice deploy" nor "chalice package" is working. I've tried the hacky solutions mentioned above like emptying out requirements.txt

Error - ModuleNotFoundError: No module named 

If I try to add this code to initial zip & then remove it later, that seem to waste the initial effort of using layer.

Please provide suggestions here or if this functionality is not yet implemented in chalice, can somebody make a note of handling this in future releases. Thank you.

**Pull request merged**